src/OrchardCore/OrchardCore.Abstractions/Json/Nodes/JNode.cs

```diff
@@ -170,26 +171,25 @@ public static bool TryParse(string json, out JsonNode? jsonNode, JsonNodeOptions
     public static string? GetNormalizedPath(this JsonNode jsonNode) => jsonNode.GetPath().GetNormalizedPath();
 
     /// <summary>
-    /// Selects a <see cref="JsonNode"/> from this <see cref="JsonObject"/> using a JSON path.
+    /// Selects a <see cref="JsonNode"/> from this <see cref="JsonObject"/> using JSONPath.
     /// </summary>
-    public static JsonNode? SelectNode(this JsonNode? jsonNode, string? path)
+    /// <param name="jsonNode">The JSON node which serves as the root of the current search..</param>
+    /// <param name="path">The JSONPath query where <c>$</c> is <paramref name="jsonNode"/>.</param>
+    /// <param name="options">Optional settings to configure the JSONPath parser.</param>
+    /// <remarks>
+    /// This method uses JsonPath.Net to evaluate the <paramref name="path"/>. For more information on JSONPath, see the
+    /// specification at https://www.rfc-editor.org/rfc/rfc9535.html or the JsonPath.Net documentation at
+    /// https://docs.json-everything.net/path/basics/.
+    /// </remarks>
+    public static JsonNode? SelectNode(this JsonNode? jsonNode, string? path, PathParsingOptions? options = null)
     {
-        if (jsonNode is null || path is null)
-        {
-            return null;
-        }
-
-        if (jsonNode is JsonObject jsonObject)
-        {
-            return jsonObject.SelectNode(path);
-        }
-
-        if (jsonNode is JsonArray jsonArray)
-        {
-            return jsonArray.SelectNode(path);
-        }
+        path = path?.Trim();
+        if (string.IsNullOrEmpty(path)) return jsonNode;
+        if (path[0] != '$') path = "$." + path;
```

Comment:
```suggestion
        if (string.IsNullOrEmpty(path)) {
            return jsonNode;
        }
        
        if (path[0] != '$') {
            path = "$." + path;
        }
```

Reply:
It's orchard core, we put open braces on new lines ;)

Reply:
Duh! lol